src/main/java/algorithms/curated170/medium/countunivaluesubtrees/CountUnivalueSubtreesDFSPassValue.java

```diff
@@ -0,0 +1,29 @@
+package algorithms.curated170.countunivaluesubtrees.medium;
+
+import algorithms.datastructures.TreeNode;
+
+public class CountUnivalueSubtreesDFSPassValue {
+
+    private int count = 0;
+
+    public int countUnivalSubtrees(TreeNode root) {
+        isUnivaluePart(root, 0);
+        return count;
+    }
+
+    private boolean isUnivaluePart(TreeNode root, int parentVal)
+    {
+        if(root == null)
+        {
+            return true;
+        }
+
+        if(!isUnivaluePart(root.left, root.val) & !isUnivaluePart(root.right, root.val))
```

Comment:
```suggestion
        if(!isUnivaluePart(root.left, root.val) | !isUnivaluePart(root.right, root.val))
```
If we fix it, it is quite fine:
![image](https://user-images.githubusercontent.com/60903744/120088663-a60b1300-c0fb-11eb-8b0c-0a095dbf6162.png)
As a not for the future audience:
`A | B`:  First, evaluate A then evaluate B. Finally, apply the OR.

 `A || B`: First, evaluate A, and only if A is false then evaluate B. Finally, apply the OR.

Reply:
Same goes for the `&` and `&&` operators. `&&` does not run and check the right side of the statement if the left side is false, because that guarantees that the expression will be false. `&&` and `||` are thus useful for things like null-checks, sanity tests and prioritizing faster operations. `&`, however, also runs the right side of the statement even if the left side is false, i.e. even when the expression is confirmed to be false. In this question, we have to DFS through all the nodes, so we need to use `&` so as to DFS to the right-side even if the left-side is not a valid subtree.